pkg/cloudprovider/provider/equinixmetal/provider.go

```diff
@@ -399,6 +399,11 @@ func (s *metalDevice) ID() string {
 	return s.device.ID
 }
 
+// TODO(xmudrii): Implement this.
+func (s *metalDevice) ProviderID() string {
+	return ""
+}
```

Comment:
Equinix Metal integrations for KKP and KubeOne are currently broken, so I wasn't able to create an Equinix Metal cluster.